python/torch_mlir/__init__.py

```diff
@@ -119,27 +118,138 @@ def like(tensor: torch.Tensor, dynamic_axes: List[int] = None):
         return TensorPlaceholder(shape, tensor.dtype)
 
 
+_example_arg = Union[TensorPlaceholder, torch.Tensor]
+_example_args_for_one_method = Union[_example_arg, Sequence[_example_arg]]
+_example_args = Union[_example_args_for_one_method, "ExampleArgs"]
+
+
+class ExampleArgs:
+    """A class representing the example arguments to an nn.Module.
+
+    In general, an nn.Module may have multiple methods that need to be compiled.
+    This requires example args for each method. This class is a lightweight
+    wrapper around a dictionary that maps method names to example arguments.
+
+    In user-facing API's, this type can always be passed interchangeably with a
+    single arg or list of args, which normalizes to an ExampleArgs for just
+    the `forward` method via the `ExampleArgs.get` static method.
+    """
+
+    def __init__(self):
+        self._example_args = {}
+
+    def add_method(self, method_name: str, example_args: _example_args_for_one_method):
+        """Adds example args for a method.
+
+        Args:
+            method_name: The name of the method. Must have not already been
+                added previously as a method.
+            example_args: The example args for the method.
+        Returns:
+            self, for chaining.
+        """
+        assert method_name not in self._example_args
+        self._example_args[method_name] = ExampleArgs._canonicalize_args(
+            example_args)
+        return self
+
+    @staticmethod
+    def get(example_args: _example_args) -> "ExampleArgs":
+        """Gets an ExampleArgs from allowed way to specify one.
```

Comment:
I'm finding the phrasing of this docstring to be a bit confusing.

Reply:
It was generated by copilot (Github AI autocomplete) lol. Fixed.